### PR TITLE
require login before viewing some pages

### PR DESCRIPTION
**Pull request by saty9**

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ export default new Router({
         header: AppHeader,
         default: Login,
         footer: AppFooter
+      },
+      props: {
+        default: true
```
Comment (KristiyanTs) [inline src/router.js:45]: what is the default for

File: src/router.js [hunk 2]
```diff
@@ -126,13 +135,42 @@ export default new Router({
...
+      return {selector: to.hash};
     } else {
-      return { x: 0, y: 0 };
+      return {x: 0, y: 0};
     }
   }
 });
+
+router.beforeEach((to, from, next) => {
+  console.log(to);
```
Comment (KristiyanTs) [inline src/router.js]: remove console.log when merging with master

File: src/router.js [hunk 3]
```diff
@@ -126,13 +135,42 @@ export default new Router({
...
+    if (router.app.$store.state.signedIn) {
+        next()
+    } else {
+      if (window.$cookies.get("jwt")) {
+        let token = window.$cookies.get("jwt");
+        let userRole = window.$cookies.get("userRole");
+        router.app.$store.commit("SIGNED_IN", [token, userRole]);
+        next()
+      } else {
+        console.log(to.fullPath);
```
Comment (KristiyanTs) [inline src/router.js]: same

File: src/views/Login.vue [hunk 4]
```diff
@@ -1,83 +1,89 @@
...
-            <div class="col-6">
-              <router-link to="/password" class="text-light">
-                <small>Forgot password?</small>
-              </router-link>
-            </div>
-            <div class="col-6 text-right">
-              <router-link to="/signup" class="text-light">
-                <small>Create new account</small>
-              </router-link>
+    <section class="section section-shaped section-lg my-0">
```
Comment (KristiyanTs) [inline src/views/Login.vue]: don't use 4 spaces for indentation because the lines become too wide

File: src/views/Login.vue [hunk 5]
```diff
@@ -88,7 +94,7 @@
       };
     },
     methods: {
-      submit: function() {
+      submit: function () {
```
Comment (KristiyanTs) [inline src/views/Login.vue]: Here you add a space between the function, which I also try to do

File: src/views/Login.vue [hunk 6]
```diff
@@ -104,16 +110,16 @@
           })
           .catch(error => this.loginFailed(error))
       },
-      loginSuccessful (response) {
+      loginSuccessful(response) {
```
Comment (KristiyanTs) [inline src/views/Login.vue]: But here

File: src/views/Login.vue [hunk 7]
```diff
@@ -104,16 +110,16 @@
...
           response.headers.authorization,
           response.data.id,
           response.data.role
         ]);
         this.$store.commit("ADD_ALERT", ["You are now logged in.", "success"]);
-        this.$router.push('/');
+        this.$router.push(this.nextUrl)
       },
-      loginFailed (error) {
+      loginFailed(error) {
```
Comment (KristiyanTs) [inline src/views/Login.vue]: and here you removed it :/

**Review by KristiyanTs — CHANGES_REQUESTED** [review]
just some inconsistencies and it wasn't really clear what default is for

File: src/router.js (hunk 1 above)
Comment (saty9) [reply src/router.js:45]: default maps to the default component (line 41) and means it takes router params as its props

File: src/router.js (hunk 2 above)
Comment (saty9) [reply src/router.js]: oops

**Pull request merged**